src/main/scala/com/redislabs/provider/redis/redisFunctions.scala

```diff
@@ -390,6 +402,49 @@ object RedisContext extends Serializable {
     conn.close()
   }
 
+  /**
+    * @param arr hashName/k/vs which should be saved in the target host
+    * @param ttl time to live(seconds)
+    */
+  def setNamedHash2(arr: Iterator[(String, String, String)], ttl: Int, redisConfig: RedisConfig,
+              readWriteConfig: ReadWriteConfig) {
+    implicit val rwConf: ReadWriteConfig = readWriteConfig
+    arr.map(h => (h._1, h)).toArray.groupBy(_._1).
```

Comment:
At the beginning, the code map with hashKey, but we found using redis pipeline well is more faster than reducing expire commands.